nix/overlays/checked-shell-script/checked-shell-script.nix

```diff
@@ -22,6 +22,14 @@ let
         ''
           #!${runtimeShell}
           set -euo pipefail
+
+          # make sure that no script creates conflicting tix files
+          # postgrest-coverage will override HPCTIXFILE to make proper use of it
+          if test ! -v HPCTIXFILE; then
+            hpctixdir=$(mktemp -d)
+            export HPCTIXFILE="$hpctixdir"/postgrest.tix
+            trap 'rm -rf $hpctixdir' EXIT
```

Comment:
This trap is overridden if the script passed in as `text` sets its own trap later. I think it would be better if we disabled hpc by default, is this feasible in a reasonable way?

Reply:
hm. I enabled the hpc flags for all builds to avoid having to rebuild everything when switching between regular commands and the coverage command. It feels a lot smoother in development.

I could just add the good old `rm -rf *.tix` as a simple solution - but I felt, it'd be good to have some kind of way to explicitely tell those commands to keep collecting .tix data.

Reply:
And once I start collecting tix files in the repo root, it'd be annoying if those are gone all of a sudden...

Reply:
Maybe we could add a flag to shell.nix to enable a hpc/coverage mode? It feels a bit messy to always have hpc on. It probably also impacts testing and later benchmarking performance, no?

Thinking of it, what would probably work here in the interim is `HPCTIXFILE=/dev/null`?

Reply:
> Maybe we could add a flag to shell.nix to enable a hpc/coverage mode? It feels a bit messy to always have hpc on.

That would be easily possible, but wouldn't really solve the problem. Even with the flag enabled, either per default or explicitly, I would still want those commands to just work without manually deleting .tix files inbetween.

> It probably also impacts testing and later benchmarking performance, no?

That's something I also thought about, but decided to defer it to later, once we're there.

> Thinking of it, what would probably work here in the interim is `HPCTIXFILE=/dev/null`?

Nope, that was the first thing I tried :). Once the file exists, the hpc-enabled-binary will try to read it and will fail if the file is empty.